pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -195,12 +198,20 @@ func (cs *ConntrackConnectionStore) AddOrUpdateConn(conn *flowexporter.Connectio
 	connKey := flowexporter.NewConnectionKey(conn)
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
-	existingConn, exists := cs.connections[connKey]
+
+	existingConn, exists := cs.Connections[connKey]
+	currTime := time.Now()
 
 	if exists {
 		existingConn.IsPresent = true
-		// avoid updating stats of the existing connection that is about to close
+		cs.expirePriorityQueue.Mutex.Lock()
 		if flowexporter.IsConnectionDying(existingConn) {
+			// If the connection is in dying state and the corresponding pqItem is already
+			// exported, then update the DyingAndDoneExport flag on the connection.
+			if existingConn.PriorityQueueItem.DyingAndDoneExport {
+				conn.DyingAndDoneExport = true
```

Comment:
This should be `existingConn` right?
If so, please make sure this is covered by unit test.